easytpl.go

```diff
@@ -203,60 +203,42 @@ func prepareTemplateTags(body string) (string, map[string][]string) {
 		return fmt.Sprintf(`{{ "%s" }}`, match)
 	})
 
-	// Remove &nbsp; inside template tags.
-
-	// Convert {%var%} to {{.var}}
-	matchFrom := 0
-	usedVars := make(map[string][]string)
-	for {
-		loc := tagsToGo.FindStringIndex(body[matchFrom:])
-		if loc == nil {
-			break
-		}
-
-		loc[0] += matchFrom
-		loc[1] += matchFrom
-		matchFrom = loc[1]
-		content := body[loc[0]:loc[1]]
-
-		// Remove &nbsp from the tag, as Squire can add that in some cases.
-		// TODO: because we use the index this mucks up the next iteration of
-		// the loop.
-		//content = findNbsp.ReplaceAllString(content, "")
-
+	usedVars := map[string][]string{}
+	// Convert {%user.firstName%} to {{.User.FirstName}}
+	body = tagsToGo.ReplaceAllStringFunc(body, func(match string) string {
 		// Allow escaping { with \{
-		if loc[0] > 0 && body[loc[0]-1] == '\\' {
-			body = body[:loc[0]-1] + content + body[loc[1]:]
-			continue
+		if match[0] == '\\' {
+			return match[1:]
 		}
 
-		tagParts := strings.Split(content[2:len(content)-2], ",")
-		tagParts[0] = strings.Title(tagParts[0])
+		submatch := tagsToGo.FindStringSubmatch(match)
+		if len(submatch) < 3 {
+			return match
+		}
 
-		// Make sure matchFrom isn't longer than the length of the string (can
-		// happen if there are multiple spaces after {%).
-		tagName := strings.Join(tagParts, ",")
-		tagNameTrimmed := strings.TrimSpace(tagName)
-		matchFrom -= len(tagName) - len(tagNameTrimmed)
+		tagParts := strings.Split(strings.TrimSpace(submatch[2]), ".")
+		if len(tagParts) < 2 {
+			return match
+		}
 
-		tag := fmt.Sprintf("{{.%s}}", tagNameTrimmed)
+		if len(tagParts[0]) > 0 {
+			tagParts[0] = strings.ToUpper(tagParts[0][0:1]) + tagParts[0][1:]
```

Comment:
Why the upper case conversion?  Just to handle the map lookup on potential duplicates?

Reply:
Our variables in the front facing template are camelcase, the backend template uses `Upper.UpperCase`. So we must convert that. The conversion here is only for the first character. `user.firstName` -> `User.FirstName` which will be correct for the templating engine.

Reply:
Got it.  I missed the string concat part there.